src/Analyzers/CSharp/CodeFixes/ConvertToRecord/ConvertToRecordEngine.cs

```diff
@@ -479,6 +479,19 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
         var documentLookup = referenceLocations.ToLookup(refLoc => refLoc.Document.Id);
         foreach (var (documentID, documentLocations) in documentLookup)
         {
+            // We don't want to process source-generated documents.  Make sure we can get back to a real document here.
+            var document = solutionEditor.OriginalSolution.GetDocument(documentID);
```

Comment:
❓ Can we not just use `refLoc.Document is SourceGeneratedDocument` (which appears to be a public type)?

Reply:
wfm.